packages/yew-macro/src/html_tree/html_component.rs

```diff
@@ -54,12 +55,46 @@ impl Parse for HtmlComponent {
             }
 
             if trying_to_close() {
-                break;
+                let cursor = input.cursor();
+                let _ = cursor
+                    .punct()
+                    .and_then(|(_, cursor)| cursor.punct())
+                    .and_then(|(_, cursor)| cursor.ident())
+                    .ok_or_else(|| {
+                        syn::Error::new(Span::call_site(), "expected a valid closing tag (e.g.: </Component>)")
+                    })?;
+
+                let fork = input.fork();
```

Comment:
Is there a reason not to adjust and use the `Parse` impl of `HtmlComponentClose`? I don't quite see where else this impl would be used if not here. Forking though looks correct, so recovery on error starts before the end tag not in the middle of it :)

This comment is ofc optional to "fix", I'd just want a reason for it.

Reply:
Because we're working with a fork of the input, not the input it self so the types didn't match. It may be possible to fix but i haven't tried to fix it.

Reply:
It could be me be stupid, but isn't fork [documented returning Self](https://docs.rs/syn/latest/syn/parse/struct.ParseBuffer.html#method.fork) ie the same type that was passed in?

Reply:
One word: lifetimes

But it could also just be intellij-rust incorrectly reporting the error (I have clippy/cargo check disabled). I didn't actually try to compile it

Reply:
Fixed